src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,310 @@
+
+
+def create_results_file_name(
+    file_format: str, output_dir: str, timestamp: str, model_name: str
+) -> str:
+    if IOFileType.CSV.value == file_format:
+        file_type = "csv"
+    elif IOFileType.XLSX.value == file_format:
+        file_type = "xlsx"
+    elif IOFileType.JSONL.value == file_format:
+        file_type = "jsonl"
+    else:
+        raise ValueError("File format is not one of: csv, xlsx, jsonl")
+
+    output_path = pathlib.Path(output_dir)
+    if not output_path.exists() or not output_path.is_dir():
+        output_dir = DEFAULTS.EVAL_DATA_DIR
+        logger.debug(
+            "Output dir provided is not a directory or does not exist, writing results to default evaluation output directory %s",
+            output_dir,
+        )
+
+    return f"{output_dir}/responses-{model_name}-{timestamp}.{file_type}"
+
+
+def get_api_base_for_model(
+    ctx: click.Context,
+    model: str,
+    temperature: float,
+    model_prompt: str,
+    max_workers: str | int | None,
+    gpus: int | None,
+    backend: str | None,
+    enable_serving_output: bool,
+) -> tuple:
+    server = None
+    if model_is_localand_valid(model):
+        logger.debug("DK-Bench model is local")
+        model_name = get_model_name(model)
+        server, api_base, _ = launch_server(
+            eval_serve=ctx.obj.config.serve,
+            tls_client_cert=ctx.params["tls_client_cert"],
+            tls_client_key=ctx.params["tls_client_key"],
+            tls_client_passwd=ctx.params["tls_client_passwd"],
+            tls_insecure=ctx.params["tls_insecure"],
+            model=model,
+            model_name=model_name,
+            max_workers=max_workers,
+            gpus=gpus,
+            backend=backend,
+            enable_serving_output=enable_serving_output,
+        )
+    else:
+        logger.debug("DK-Bench model is an endpoint")
+        http_params = http_client(
+            {
+                "tls_client_cert": ctx.params["tls_client_cert"],
+                "tls_client_key": ctx.params["tls_client_key"],
+                "tls_client_passwd": ctx.params["tls_client_passwd"],
+                "tls_insecure": ctx.params["tls_insecure"],
+            }
+        )
+
+        api_key = os.environ.get("STUDENT_ENDPOINT_API_KEY", None)
+        if api_key is None:
+            logger.debug(
+                "API_KEY for model at endpoint %s is not set. To set it set the environment variable $STUDENT_ENDPOINT_API_KEY",
+                model,
+            )
+
+        model_name = get_endpoint_model_name(model, api_key, http_params)
+        api_base = model
+
+    openai_client = get_openai_client(model_api_base=api_base, api_key=api_key)
+    model_config = ModelConfig(
+        model_name=model_name, temperature=temperature, system_prompt=model_prompt
+    )
+
+    return server, model_config, openai_client
+
+
+def make_run_dir(output_dir: str) -> str:
+    now = datetime.now()
+    timestamp = now.strftime("%m-%d-%Y_%H-%M-%S-%f")
```

Comment:
It seems like we are using some magical string formatting here. If you don't want to use the built-in string formatting then could you please leave a comment explaining why we're doing it this way?

Reply:
What's the default format for a timestamp that you mentioned below?

If the user runs this bench mark over and over I didn't want results files piling up in DEFAULTS.EVAL_DIR, so I'm creating a directory with the timestamp of the run in the name. I wanted to have a timestamp that goes down to the microsecond level just so that two runs couldn't possibly have the same directory.

Reply:
@alimaredia You can use the `datetime.isoformat()` method if you're looking for a string. Otherwise you can use `datetime.timestamp` to get a floating point representation of the current UNIX timestamp. 

```suggestion
    timestamp = now.isoformat()
```